alchemy/src/prisma/connection.ts

```diff
@@ -0,0 +1,274 @@
+
+/**
+ * Properties for creating a Prisma database connection
+ */
+export interface ConnectionProps extends PrismaApiOptions {
+  /**
+   * The project that the database belongs to
+   */
+  project: string | Project;
+
+  /**
+   * The database to create a connection for
+   */
+  database: string | Database;
+
+  /**
+   * Name of the connection
+   */
+  name: string;
+}
+
+/**
+ * Output returned after Prisma database connection creation
+ */
+export interface Connection extends Resource<"prisma::Connection"> {
+  /**
+   * The ID of the connection
+   */
+  id: string;
+
+  /**
+   * The ID of the project
+   */
+  projectId: string;
+
+  /**
+   * The ID of the database this connection belongs to
+   */
+  databaseId: string;
+
+  /**
+   * Name of the connection
+   */
+  name: string;
+
+  /**
+   * Database connection string (sensitive)
+   */
+  connectionString: Secret;
+
+  /**
+   * Time at which the connection was created
+   */
+  createdAt: string;
+}
+
+/**
+ * Creates a Prisma database connection string for accessing a database.
+ *
+ * @example
+ * ## Create a database connection
+ *
+ * ```ts
+ * const connection = await Connection("app-connection", {
+ *   project: project,
+ *   database: database,
+ *   name: "app-production"
+ * });
+ * ```
+ *
+ * @example
+ * ## Create connection with explicit IDs
+ *
+ * ```ts
+ * const connection = await Connection("backup-connection", {
+ *   project: "project-123",
+ *   database: "database-456",
+ *   name: "backup-reader"
+ * });
+ * ```
+ *
+ * @example
+ * ## Access connection string
+ *
+ * ```ts
+ * const connection = await Connection("my-connection", {
+ *   project: project,
+ *   database: database,
+ *   name: "web-app"
+ * });
+ *
+ * // Use the connection string in your application
+ * console.log(await connection.connectionString.unencrypted);
+ * ```
+ */
+export const Connection = Resource(
+  "prisma::Connection",
+  async function (
+    this: Context<Connection>,
+    id: string,
+    props: ConnectionProps,
+  ): Promise<Connection> {
+    const api = createPrismaApi(props);
+    const projectId =
+      typeof props.project === "string" ? props.project : props.project.id;
+    const databaseId =
+      typeof props.database === "string" ? props.database : props.database.id;
+    const connectionId = this.output?.id;
+
+    if (this.phase === "delete") {
+      try {
+        if (connectionId) {
+          await deleteConnection(api, projectId, databaseId, connectionId, id);
+        }
+      } catch (error) {
+        logger.error(`Error deleting Prisma connection ${id}:`, error);
+        throw error;
+      }
+      return this.destroy();
+    }
+
+    try {
+      let connection: any;
+
+      if (this.phase === "update" && connectionId) {
+        // Connections are immutable once created, so just return current state
+        connection = await getConnection(
+          api,
+          projectId,
+          databaseId,
+          connectionId,
+        );
```

Comment:
We should check this.props vs props and error on any "invalid updates" rather than succeed silently? If it's immutable, changes should be errors